staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -244,13 +244,15 @@ func (cnc *CloudNodeController) updateNodeAddress(ctx context.Context, node *v1.
 		}, nil
 	}
 
-	instanceMeta, err := instanceMetadataGetter(node.Spec.ProviderID, node)
-	if err != nil {
-		utilruntime.HandleError(err)
-		return
+	if instanceMetadata == nil {
```

Comment:
Thinking that the nil case should always result in an error and we should ensure that `getInstanceMetadata` is called prior to calling `updateNodeAddress`.

Reply:
Updated, maybe better now.